internal/core/src/storage/MmapChunkManager.cpp

```diff
@@ -0,0 +1,240 @@
+namespace {
+static constexpr int kMmapDefaultProt = PROT_WRITE | PROT_READ;
+static constexpr int kMmapDefaultFlags = MAP_SHARED;
+};  // namespace
+
+// todo(cqy): After confirming the append parallelism of multiple fields, adjust the lock granularity.
+
+MmapBlock::MmapBlock(const std::string& file_name,
+                     const uint64_t file_size,
+                     BlockType type)
+    : file_name_(file_name), file_size_(file_size), block_type_(type) {
+    allocated_size_.fetch_add(file_size);
+    // create tmp file
+    int fd = open(file_name_.c_str(), O_RDWR | O_CREAT, S_IRUSR | S_IWUSR);
+    if (fd == -1) {
+        LOG_ERROR("Failed to open mmap tmp file");
+        return;
+    }
+    // append file size to 'file_size'
+    if (lseek(fd, file_size - 1, SEEK_SET) == -1) {
+        LOG_ERROR("Failed to seek mmap tmp file");
+        return;
+    }
+    if (write(fd, "", 1) == -1) {
+        LOG_ERROR("Failed to write mmap tmp file");
+        return;
+    }
+    // memory mmaping
+    addr_ = static_cast<char*>(
+        mmap(nullptr, file_size, kMmapDefaultProt, kMmapDefaultFlags, fd, 0));
+    if (addr_ == MAP_FAILED) {
+        LOG_ERROR("Failed to mmap");
+        return;
+    }
+    offset_.store(0);
+}
+
+MmapBlock::~MmapBlock() {
+    if (addr_ != nullptr) {
+        munmap(addr_, file_size_);
+    }
+    if (std::ifstream(file_name_.c_str()).good()) {
+        std::remove(file_name_.c_str());
+    }
+    allocated_size_.fetch_sub(file_size_);
+}
+
+void*
+MmapBlock::Get(const uint64_t size, ErrorCode& error_code) {
+    if (file_size_ - offset_.load() < size) {
+        error_code = ErrorCode::MemAllocateSizeNotMatch;
+        return nullptr;
+    } else {
+        error_code = ErrorCode::Success;
+        return (void*)(addr_ + offset_.fetch_add(size));
+    }
+}
+
+MmapBlockPtr
+MmapBlocksHandler::AllocateFixSizeBlock(ErrorCode& error_code) {
+    if (fix_size_blocks_cache_.size() != 0) {
+        // return a mmap_block in fix_size_blocks_cache_
+        auto block = std::move(fix_size_blocks_cache_.front());
+        std::cout << "find a block in cache " << block.get() << std::endl;
+        fix_size_blocks_cache_.pop();
+        error_code = ErrorCode::Success;
+        return std::move(block);
+    } else {
+        // if space not enough for create a new block, clear cache and check again
+        if (fix_mmap_file_size_ + Size() > max_disk_limit_) {
+            error_code = ErrorCode::MmapSpaceInsufficient;
+            return nullptr;
+        }
+        std::cout << "generate a new block " << std::endl;
+        error_code = ErrorCode::Success;
+        return std::move(std::make_unique<MmapBlock>(
+            GetMmapFilePath(), GetFixFileSize(), MmapBlock::BlockType::Fixed));
+    }
+}
+
+MmapBlockPtr
+MmapBlocksHandler::AllocateLargeBlock(const uint64_t size,
+                                      ErrorCode& error_code) {
+    if (fix_mmap_file_size_ + Capacity() > max_disk_limit_) {
```

Comment:
is it a good idea to align with fixed block size? for example, always allocate 8M if want 7M